identity/src/error.rs

```diff
@@ -150,3 +120,30 @@ impl Error for SigningError {
         self.source.as_ref().map(|s| &**s as &dyn Error)
     }
 }
+
+#[derive(Debug)]
+/// Error produced when trying to convert `libp2p_identity::Keypair` to a more concrete keypair failed.
```

Comment:
We typically have these the other way round:

```suggestion
/// Error produced when trying to convert `libp2p_identity::Keypair` to a more concrete keypair failed.
#[derive(Debug)]
```